tests/e2e/vsan_stretched_cluster_utils.go

```diff
@@ -361,3 +374,141 @@ func createPodsInParallel(client clientset.Interface, namespace string, pvclaims
+	defer wg.Done()
+	for _, pv := range persistentVolumes {
+		framework.Logf(fmt.Sprintf("Updating labels %+v for pv %s in namespace %s",
+			labels, pv.Name, namespace))
+		pv.Labels = labels
+		_, err := client.CoreV1().PersistentVolumes().Update(ctx, pv, metav1.UpdateOptions{})
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+	}
+}
+
+// getMasterIpOnSite returns IP address of master node on a particular site. This function has to be
+// only used while testing distributed - SEB testbed deployment model.
+func getMasterIpOnSite(ctx context.Context, client clientset.Interface, primarySite bool) (string, error) {
+	siteEsxMap := make(map[string]bool)
+	masterIpOnSite := ""
+
+	siteHosts := fds.secondarySiteHosts
+	if primarySite {
+		siteHosts = fds.primarySiteHosts
+	}
+
+	for _, x := range siteHosts {
+		siteEsxMap[x] = true
+	}
+	allMasterIps := getK8sMasterIPs(ctx, client)
+	framework.Logf("all master ips : %v", allMasterIps)
+	sshClientConfig := &ssh.ClientConfig{
+		User: "root",
+		Auth: []ssh.AuthMethod{
+			ssh.Password(k8sVmPasswd),
+		},
+		HostKeyCallback: ssh.InsecureIgnoreHostKey(),
+	}
+	framework.Logf("Site esx map : %v", siteEsxMap)
+	// Assuming atleast one master is on that site
+	vcAddress := e2eVSphere.Config.Global.VCenterHostname
+	for _, masterIp := range allMasterIps {
+		cmd := "export GOVC_INSECURE=1;"
+		cmd += fmt.Sprintf("export GOVC_URL='https://administrator@vsphere.local:Admin!23@%s';", vcAddress)
+		cmd += fmt.Sprintf("govc vm.info --vm.ip=%s;", masterIp)
+		result, err := sshExec(sshClientConfig, masterIp, cmd)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		hostIp := strings.Split(result.Stdout, "Host:")
+		host := strings.TrimSpace(hostIp[1])
+		if siteEsxMap[host] {
+			masterIpOnSite = masterIp
+			break
+		}
+	}
+	framework.Logf("Master IP on site : %s", masterIpOnSite)
+	if masterIpOnSite != "" {
+		return masterIpOnSite, nil
+	} else {
+		return "", fmt.Errorf("couldn't find a master running on site")
+	}
+}
+
+// changeLeaderOfContainerToComeUpOnMaster ensures that the leader of a container comes up on
+// a specific master node on that site
+func changeLeaderOfContainerToComeUpOnMaster(ctx context.Context, client clientset.Interface,
+	sshClientConfig *ssh.ClientConfig, csiContainerName string, primarySite bool) error {
+	// Fetch the IP address of master node on that site
+	masterIpOnSite, err := getMasterIpOnSite(ctx, client, primarySite)
+	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+	// Get the master Ip where leader of csi container is running
+	allMasterIps := getK8sMasterIPs(ctx, client)
+	// Remove master ip which is on that site from list of master ip of all nodes
+	for i, v := range allMasterIps {
+		if v == masterIpOnSite {
+			allMasterIps = append(allMasterIps[:i], allMasterIps[i+1:]...)
+			break
+		}
+	}
+
+	leaderFoundOnsite := false
+	waitErr := wait.PollImmediate(healthStatusPollInterval, pollTimeout, func() (bool, error) {
```

Comment:
do we need this

Reply:
Yes, we need this as leader might not come up on the specific master node on first try. If it doesn't come up, we pause and kill that container on other master nodes. Hence, polling and checking if it comes up or not.